### PR TITLE
Hex Strings

### PR DESCRIPTION
**Pull request by lthiery**

### CODE REVIEW

**Review by plaes — APPROVED** [review]

**Comment by plaes** [comment]
You may also want to revise copyright strings...

**Review by lucasgranberg — APPROVED** [review]

**Comment by lthiery** [comment]
On second though, I realized feature-gating from_str was not terribly useful, so I've reorganized things so that from_str is always available.

File: lorawan-encoding/src/string.rs [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+                hex::decode_to_slice(s.as_bytes(), &mut res)?;
+                Ok(Self::from(res))
+            }
+        }
+
+        #[cfg(feature = "with-to-string")]
+        impl std::string::ToString for $type {
+            fn to_string(&self) -> std::string::String {
+                let mut res = std::string::String::with_capacity($size * 2);
+                res.extend(std::iter::repeat('-').take($size * 2));
```
Comment (plaes) [inline lorawan-encoding/src/string.rs:27]: Why `-`?
Is there a test with under- and oversized hex string?

Comment (lthiery) [reply lorawan-encoding/src/string.rs:27]: We never assert size, but the `-` ensure an unwritten char would be obvious.

Comment (lthiery) [reply lorawan-encoding/src/string.rs:27]: Concerning missized strings, good point! I'll work on that now

Comment (lthiery) [reply lorawan-encoding/src/string.rs:27]: The hex library handles them nicely it seems

**Pull request merged**